File: practice/convert/example2_book_library/book-library/src/main/java/dev/starryeye/book_library/controller/user/ChangeUsernameController.java
package dev.starryeye.book_library.controller.user;

import dev.starryeye.book_library.application.command.ChangeUsernameService;
import dev.starryeye.book_library.controller.user.request.ChangeUsernameRequest;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class ChangeUsernameController {

    private final ChangeUsernameService service;

    @PutMapping("/api/v1/users/change-name")
    public void changeUsername(@Valid @RequestBody ChangeUsernameRequest request) {
        service.change(request.toCommand());
    }
}

File: practice/convert/example2_book_library/kotlin/book-library/src/main/kotlin/dev/starryeye/book_library/controller/user/ChangeUsernameController.kt
package dev.starryeye.book_library.controller.user

import dev.starryeye.book_library.application.user.command.ChangeUsernameService
import dev.starryeye.book_library.controller.user.request.ChangeUsernameRequest
import dev.starryeye.book_library.controller.user.request.toCommand
import jakarta.validation.Valid
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class ChangeUsernameController(
    private val service: ChangeUsernameService,
) {

    @PutMapping("/api/v1/users/change-name")
    fun changeUsername(
        @Valid @RequestBody request: ChangeUsernameRequest,
    ) {
        service.changeUsername(request.toCommand())
    }
}